spec/ruby/language/string_spec.rb

```diff
@@ -260,6 +260,16 @@ def long_string_literals
     it "produce different objects for literals with the same content in different files if they have different encodings" do
       ruby_exe(fixture(__FILE__, "freeze_magic_comment_across_files_diff_enc.rb")).chomp.should == "true"
     end
+
+    it "interpolated string is frozen" do
+      ruby_exe(fixture(__FILE__, "freeze_magic_comment_interpolation.rb")).chomp.should == "true"
+    end
+
+    describe "when disabled" do
+      it "interpolated string is not frozen" do
+        ruby_exe(fixture(__FILE__, "non_frozen_interpolation.rb")).chomp.should == "false"
+      end
+    end
```

Comment:
There are already specs at https://github.com/oracle/truffleruby/blob/73b406475d50944fbcb7913b3da9ad2ccdd34faf/spec/ruby/language/string_spec.rb#L295-L310 and they use `eval` which is much faster than creating a subprocess for this.
They need an extra `it` though, like:
```ruby
  ruby_version_is ""..."3.0" do
    it "creates a frozen String when # frozen-string-literal: true is used" do
      code = <<~'RUBY'
      # frozen-string-literal: true
      "a#{6*7}c"
      RUBY
      eval(code).should.frozen?
    end
  end
```

Reply:
alt-romes:master